lms/djangoapps/course_api/blocks/transformers/student_view.py

```diff
@@ -72,6 +74,23 @@ def transform(self, usage_info, block_structure):
         """
         Mutates block_structure based on the given usage_info.
         """
+
         for block_key in block_structure.post_order_traversal():
             if block_structure.get_xblock_field(block_key, 'category') not in self.requested_student_view_data:
                 block_structure.remove_transformer_block_field(block_key, self, self.STUDENT_VIEW_DATA)
+            student_view_data = block_structure.get_transformer_block_field(
```

Comment:
Can you put this block of code under an `else` clause?  As this code doesn't need to be run if `html` blocks were not requested.

Reply:
Addressed by https://github.com/edx/edx-platform/pull/15905/commits/86235d261e710f16a4790a761e21b604e4883342#diff-1a30c64335d7765247aec974a4d8601eR81

Reply:
@nasthagiri CC @bradenmacdonald

With d3fbda1, I've removed the `%%USER_ID%%` substitution step from the `StudentViewTransformer.transform()`, because I was getting test failures after the feature flag addition, and I wasn't happy with the duplicated code and very `HtmlBlock`-specific code that was added here.

I wasn't able to sort it out quickly, and since [you said it wasn't a blocker for merging](https://github.com/edx/edx-platform/pull/15905#discussion_r142288178), I had to let it go.

But there has to be a nice way to do it -- maybe by introducing a decorator on `student_view_data`, like `@XBlock.supports("usage_info_context")`, and calling `student_view_data` again with `context=usage_info`?  Not sure, but it requires more thought than I have time for right now.

Sorry for any confusion on this!